editor/src/Components/CrashNavigationSearchForm.js

```diff
@@ -1,41 +1,144 @@
-import React, { useState } from "react";
+import React, { useState, useCallback, useEffect } from "react";
 import { useHistory } from "react-router-dom";
-import { Form, Input, InputGroup, InputGroupAddon, Button } from "reactstrap";
+import { useLazyQuery } from "@apollo/react-hooks";
+import { gql } from "apollo-boost";
+import {
+  Form,
+  FormFeedback,
+  Input,
+  InputGroup,
+  InputGroupAddon,
+  Button,
+  InputGroupButtonDropdown,
+  DropdownToggle,
+  Spinner,
+} from "reactstrap";
+import { appCodeName } from "../helpers/environment";
 
-// captures any non-number except the `t` and `T` (for temp crashes)
-const CRASH_ID_EXCLUDE_CHARACTERS_REGEX = /[^\dtT]*/g;
+const DEFAULT_GLOBAL_SEARCH_FIELD = "record_locator";
+const localStorageKey = `${appCodeName}_global_search_field`;
+
+const CRASH_QUERY = gql`
+  query CrashNavigationSearch($searchTerm: String!) {
+    record_locator: crashes(
+      limit: 1
+      where: {
+        record_locator: { _eq: $searchTerm }
+        is_deleted: { _eq: false }
+      }
+    ) {
+      id
+      record_locator
+    }
+    case_id: crashes(
+      limit: 1
+      where: { case_id: { _eq: $searchTerm }, is_deleted: { _eq: false } }
+    ) {
+      id
+      record_locator
+    }
+  }
+`;
 
 const CrashNavigationSearchForm = () => {
-  const [crashSearchId, setCrashSearchId] = useState("");
-  let history = useHistory();
+  // Stores the search input that will be passed to the crash query
+  const [searchTerm, setSearchTerm] = useState("");
+  // Keeps track of if we have searched the current search term
+  const [hasSearchedTerm, setHasSearchedTerm] = useState(false);
+  // Setting that controls if we are searching by `record_locator` or `case_id`
+  const [globalSearchField, setGlobalSearchField] = useState(
+    localStorage.getItem(localStorageKey) || DEFAULT_GLOBAL_SEARCH_FIELD
+  );
+
+  const [searchForCrash, { loading, data }] = useLazyQuery(CRASH_QUERY);
+
+  const history = useHistory();
+
+  const changeSearchField = useCallback(() => {
+    /**
+     * Control the search field name and keep it in sync with local storage
+     */
+    setGlobalSearchField(prevState => {
+      const newGlobalSearchField =
+        prevState === "record_locator" ? "case_id" : "record_locator";
+      localStorage.setItem(localStorageKey, newGlobalSearchField);
+      return newGlobalSearchField;
+    });
+    setHasSearchedTerm(false);
+  }, []);
+
+  const retrievedRecordLocator = data?.[globalSearchField]?.[0]?.record_locator;
+
+  /**
+   * Hook that redirects to crash page once we have found a
+   * crash record
+   */
+  useEffect(() => {
```

Comment:
I was able to replicate the bug that Mateo found, and I also think that it is an edge case that might lead to this `useEffect` taking on more work to resolve. I thought about how to tie the state updates more closely to the request trigger, and I remembered that we ran into issues with the [`onCompleted`](https://www.apollographql.com/docs/react/data/queries#queryhookoptions-interface-oncompleted) callback not working as expected in Moped.

[Rose wrote this snazzy code](https://github.com/cityofaustin/atd-moped/blob/46e3edc09bb901286ba9fc75551f494838ff324c/moped-editor/src/views/projects/projectsListView/useProjectListViewQuery/useCsvExport.js#L193-L204) that is a nice workaround for getting a solid chain of events after the async call. Might be helpful for the next iteration.

Reply:
thanks, Mike! (and Mateo for testing, too). i was just coming here to say that i think using the imperative `client` to make queries is the only way this feature is going to work. awesome to have this example from Rose 🙌 

looks like a lot of other folks have run into this issue, too, e.g. [this SO post](https://stackoverflow.com/questions/66020910/apollo-is-it-possible-to-clear-the-data-obj-after-calling-uselazyquery).